Meetups/26/2025-03-27-minutes.md

```diff
@@ -0,0 +1,186 @@
+
+Henk: I have started with MQTT and myzone. Then went to NGSI-LD.
+
+Henk: I assume the IoT devices can be anything. I'll get a bit deeper into that in a minute. It doesn't have to be a WoT device. In fact, the assumption is that most of them are not.
+
+Henk: Consumers do not talk to the devices directly. They talk to a digital twin.
+
+Henk: You can add services.
+
+Henk: This is more secure than the servers. Also it is a bit against the current design of WoT. This way, the Consumer are talking to a single endpoint.
+
+Henk: Security is a real issue. Not good encryption, weak auth mechanisms. It is good that WoT is going around this a bit.
+
+Henk: A casino database was hacked through a smart fish tank thermometer.
+
+Henk: In HiveOT, Things do not run servers, if they do they get secured behind a firewall. This greatly reduces the attack surface.
+
+Henk: By having a hub, you can centralize authentication and authorization. FW updates can be centralized. Interaction experience is uniform.
+
+Henk: WoT devices can be integrated easily, as well as many other devices. Everything is exposed as a WoT device again.
+
+Henk: We have added some plugins as well. Like history information etc.
+
+Henk: The DT router has interesting features. You can add routers to external devices or create TDs for services.
+
+Henk: (shows the action flow)
+
+Henk: The router can do validation like access right checking. Actions can be kept as a queue.
+
+Henk: (shows event flow, also valid for property updates)
+
+Henk: All messages have an envelope, like message id, timestamp etc.
+
+Henk: You can actually use the bindings independent from the hub.
+
+Henk: Current services are: Launcher, history of events, web ui, server CA management, and more.
+
+Henk: Protocol bindings include zwave, insteon, 1-wire
+
+Henk: The command line has some utilities. You can generate client code and also vocabularies.
+
+Henk: For the demo now. This is an HTMX based UI.
+
+Henk: You can click on a property and see history.
+
+Henk: (shows a TD)
+
+Henk: I know that this is not a compliant TD. The forms are omitted.
+
+Henk: Without type annotations, I don't think that the standards will go far.
+
+Henk: I was struggling with enums . I am now going with oneOf so that I can add title.
+
+Henk: HiveOT also has a directory. It shows services and Things, based on tags developed for HiveOT.
+
+Henk: Some devices have a lot of properties. Like this multi sensor having 120. The TD is massive and that is without forms.
+
+Henk: So you can ask how we can interact with Things without forms.
+
+Henk: I want to put them but basically you are interacting with all the Things over the hub. So the expressive power of forms is not needed. We can assume.
+
+Henk: The hub is meant for administrator people. You can see running services for example.
```

Comment:
The hubcli commandline utility is meant for administrators.

Reply:
```suggestion
Henk: The hubcli commandline utility is meant for administrators. You can see running services for example.
```